### PR TITLE
B 21374 ub fields int

### PR DESCRIPTION
**Pull request by cameroncaci**
> [!IMPORTANT]
> This ticket has a LOT of AC. Please do not rush this review.

## Summary

Introduces the optional and OCONUS specific fields of accompanied tour and dependent under/over 12 years of age. Feature flags the frontend portion of the application - the backend is not FF'd as it is the internal API and optional, nullable fields with no intent to ever make not nullable. This means the backend works as intended with or without the FF. The Prime cannot control these fields at this time, this is purely for customer and office user.

These OCONUS specific fields are for UB allowance enabling.

## How to test

Please test with FF on and off

### Setup
1. Think of two different duty locations
2. Once you have two different duty locations, open one up within the database, navigate to its address, and set the address `is_oconus` to true
3. Now you have a CONUS and OCONUS duty location
4. Update `.envrc` and adjust `+export FEATURE_FLAG_UNACCOMPANIED_BAGGAGE=true`

### Customer
1. Navigate to milmovelocal and create a new account
2. Create a new order and fill out all forms until you reach "Tell us about your move orders"
image
3. Fill in current and new duty location, ensuring to set one of these locations to the OCONUS version you created in the "Setup" section above.
5. Test adding dependents
6. Remove your OCONUS duty location and check that the OCONUS fields dissappear
7. Try to break it
8. Fully submit your move
9. Review your request and make sure you can edit your orders
10. Try that break that part too
11. Proceed to Office

### Office
1. Once you have submitted the move as a customer, login to Office and preview the submitted move as a SC and TOO
2. The SC should be able to see the information under entitlements and update them
3. The TOO should be able to see the information under entitlements and update them
4. These OCONUS entitlement fields shouldn't be present on CONUS moves, verify that
5. Try to break it

## Recordings
https://github.com/user-attachments/assets/1d8f5474-5b7c-4caa-a20c-77e73c3df358

https://github.com/user-attachments/assets/480c886f-7ee9-4afa-9d0d-a1b9f84ea6f1

## 508 Screenshot
Tooltips do not violate ANDI
image

### CODE REVIEW

File: migrations/app/schema/20241018091722_oconus-entitlements-enhancement.up.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+ALTER TABLE entitlements
+ADD COLUMN accompanied_tour BOOLEAN NULL,
+    ADD COLUMN dependents_under_twelve INTEGER NULL,
```
Comment (begrohmann) [inline migrations/app/schema/20241018091722_oconus-entitlements-enhancement.up.sql:4]: Need ADD COLUMN IF NOT EXISTS on these

Comment (cameroncaci) [reply migrations/app/schema/20241018091722_oconus-entitlements-enhancement.up.sql]: resolved in b18166514bfa46643e890b9c4f7e968dea6f6f20

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
1. From the AC - "When I select yes,
I can enter values for the number of dependents "12 and Older" AND "Under 12" (clear out values entered when the user changes dependents included answer back to "No").
I wasn't seeing the number of dependents under the age of 12 and over 12 clearing out when toggling the "Are dependents included in your orders" radio buttons - values persisted

2. I see 0s for dependents < 12 and >= 12 when viewing the Edit Allowances page as an SC:
image
Steps I took:
Viewed allowances on move summary page as SC
Viewed allowances on move summary page as TOO
Edited allowances as TOO (which worked great!)
Tried to edit allowances as SC and saw the 0 values

3. This is just a thought and not necessarily requesting you change this - I could see Dan not liking the placement of these tooltips and he might prefer them to be more inline with the text 


All AC except the last one and the one I mentioned above look like they've been met!
Seeing the entitlement DB values match the UI inputs, as well as get updated based on the TOO editing them.

**Comment by TevinAdams** [comment]
I was able to change the Number of dependents under age of 12 as a SC. Not sure if that's correct or not because the testing instructions says SC should be able to see the information and TOO should be able to update it.

**Comment by cameroncaci** [comment]
Converting back to draft to address changes and introduce new AC. New functionality will be as follows:

- Customer being able to edit accompanied tour, dependents under twelve, dependents twelve and over on their orders after it has been submitted
- Customer being able to preview these UB fields when selecting "Review request"
- SC being able to edit UB fields
- Checking with PO on tooltip preview

**Comment by cameroncaci** [comment]
> 1. From the AC - "When I select yes,
>    I can enter values for the number of dependents "12 and Older" AND "Under 12" (clear out values entered when the user changes dependents included answer back to "No").
>    I wasn't seeing the number of dependents under the age of 12 and over 12 clearing out when toggling the "Are dependents included in your orders" radio buttons - values persisted

@traskowskycaci I did this intentionally so that if the user is just clicking around, they can keep their values without having to re-type information they've already entered - figured it'd be a smoother experience. I have conditionals on the submit that will default these to null if dependents are set to false/('no'). Let me know if this suffices and if it works as intended

**Comment by cameroncaci** [comment]
> I was able to change the Number of dependents under age of 12 as a SC. Not sure if that's correct or not because the testing instructions says SC should be able to see the information and TOO should be able to update it.

The AC required us to fill in the gaps here in there which is how we got some dev proposed AC. SC should be able to edit allowances and is fully able to now - ready for review again (Once I verify the pipeline is passing)

**Comment by cameroncaci** [comment]
Failing test is flaky

**Comment by cameroncaci** [comment]
Failing tests are flaky

**Review by danieljordan-caci — COMMENTED** [review]
Is there a reason the customer can't edit their information after adding their orders but before submitting their shipments?

Screenshot 2024-10-25 at 3 14 27 PM

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Screenshot 2024-10-25 at 3 36 06 PM

Some UI requested changes... Can you shorten these a bit? Suggestions:
- `Accompanied tour` with a `Yes` or `No` value instead of `Is this an accompanied tour?`
- `Dependents under age 12` instead of `Number of dependents under the age of 12`
- `Dependents over age 12` instead of `Number of dependents of the age 12 or over`

And you could keep the extended description in this part
Screenshot 2024-10-25 at 3 38 28 PM


Also could you please update/add the storybooks for the `EditOrdersForm`, `AddOrdersForm` and the `AllowancesList` form?

And... Don't want to open a can here, but remember the SC can add orders for a service member...
Screenshot 2024-10-25 at 3 42 55 PM

**Review by begrohmann — APPROVED** [review]

**Comment by traskowskycaci** [comment]
> > 1. From the AC - "When I select yes,
> >    I can enter values for the number of dependents "12 and Older" AND "Under 12" (clear out values entered when the user changes dependents included answer back to "No").
> >    I wasn't seeing the number of dependents under the age of 12 and over 12 clearing out when toggling the "Are dependents included in your orders" radio buttons - values persisted
> 
> @traskowskycaci I did this intentionally so that if the user is just clicking around, they can keep their values without having to re-type information they've already entered - figured it'd be a smoother experience. I have conditionals on the submit that will default these to null if dependents are set to false/('no'). Let me know if this suffices and if it works as intended

Morning! Honestly, I like your implementation. However, it would probably be good to run it by Dan that AC mentioning clearing out the values did come from the him & his team

**Comment by cameroncaci** [comment]
Moving back to draft until new AC gets added in

**Comment by cameroncaci** [comment]
> I was able to change the Number of dependents under age of 12 as a SC. Not sure if that's correct or not because the testing instructions says SC should be able to see the information and TOO should be able to update it.

Customer confirms this is desired capabilities

**Comment by cameroncaci** [comment]
> > > 1. From the AC - "When I select yes,
> > >    I can enter values for the number of dependents "12 and Older" AND "Under 12" (clear out values entered when the user changes dependents included answer back to "No").
> > >    I wasn't seeing the number of dependents under the age of 12 and over 12 clearing out when toggling the "Are dependents included in your orders" radio buttons - values persisted
> > 
> > 
> > @traskowskycaci I did this intentionally so that if the user is just clicking around, they can keep their values without having to re-type information they've already entered - figured it'd be a smoother experience. I have conditionals on the submit that will default these to null if dependents are set to false/('no'). Let me know if this suffices and if it works as intended
> 
> Morning! Honestly, I like your implementation. However, it would probably be good to run it by Dan that AC mentioning clearing out the values did come from the him & his team

Thank you for pointing that out, asked and concurred (See agility for POA)

**Comment by cameroncaci** [comment]
> Screenshot 2024-10-25 at 3 36 06 PM
> 
> Some UI requested changes... Can you shorten these a bit? Suggestions:
> 
> * `Accompanied tour` with a `Yes` or `No` value instead of `Is this an accompanied tour?`
> * `Dependents under age 12` instead of `Number of dependents under the age of 12`
> * `Dependents over age 12` instead of `Number of dependents of the age 12 or over`
> 
> And you could keep the extended description in this part Screenshot 2024-10-25 at 3 38 28 PM
> 
> Also could you please update/add the storybooks for the `EditOrdersForm`, `AddOrdersForm` and the `AllowancesList` form?
> 
> And... Don't want to open a can here, but remember the SC can add orders for a service member... Screenshot 2024-10-25 at 3 42 55 PM

Proposed to customer and added as new AC, will action

**Review by TevinAdams — APPROVED** [review]
Tested this and it works as expected.

**Review by traskowskycaci — APPROVED** [review]
Looks good

**Review by danieljordan-caci — APPROVED** [review]

**Pull request merged**